### ISSUE TITLE
change land color to transparent

### ISSUE THREAD

**Issue by Marston** [lukelbd/proplot#290]
I would like to plot the coastlines but not to have the data over the land masked out. Can I change the land fill to transparent?

**Comment by Marston** [comment]
Figured it out. Use coast=True

**Comment by lukelbd** [comment]
Cool. Your title also made me realize it might be useful to have the ability to change the opacity `alpha` of different geographic features so I added a bunch of rc settings: `rc['land.alpha']`, `rc['ocean.alpha']`, `rc['coast.alpha']`, etc.

Here's an example (below is the same as passing `landalpha` to `ax.format()`):

```python
import proplot as pplt
fig = pplt.figure()
ax = fig.subplot(proj='cyl', land=True, coast=True, landalpha=0.2)
```

![tmp](https://user-images.githubusercontent.com/19657652/136202311-e24c226b-49f5-4439-8575-c06de01cfc7a.png)

**Comment by Marston** [comment]
I am not able to pass landalpha to the subplot or subplots routines. Are the changes you made pushed to the code repo and released as an update? Otherwise, this is not working for me. On another note, I keep getting this error that I don't understand: 
`
C:\Users\AppData\Local\Temp/ipykernel_3600/530739346.py:28: ProPlotWarning: Unexpected object AxesImage(70.1389,1356.25;1923.61x616.806) passed to _apply_edgefix.
  em = eax.pcolorfast(ev, cmap='gist_ncar_r', vmin=0.5, vmax=50.0)
`\
In addition, I getting as default discrete colorbars which does not work when I'm plotting data that are symmetric around zero. Proplot doesn't show the white color for 0s and values close to it for the colormap: seismic, e.g.
How can I remove the discrete colorbar feature?

**Comment by lukelbd** [comment]
Yep the `landalpha` feature is not yet in a release -- it's just on the "master" branch (which itself can be installed with `pip install git+https://github.com/lukelbd/proplot`). Releases happen when there's enough changes to justify it.

Thanks for alerting me to that `pcolorfast` issue, that was a one-line fix (ad1f00ff). It will no longer trigger warnings in the next version (note that warning message was harmless -- just annoying/unnecessary).

That colormap issue is some really weird/insiduous issue with cartopy projection transforms. Someone else emailed me with the same problem -- it's also possible to encounter this with just cartopy + matplotlib (i.e. without proplot).  For now, if you try using a different map projection, or if you try tweaking the edge coordinates of your dataset (for example, setting the maximum latitude to 89 instead of 90, or setting the maximum longitude to 179 instead of 180), you should be able to fix it. But the behavior is pretty unpredictable, seems to depend on the dataset and possibly even the figure size.

**Comment by lukelbd** [comment]
Also, just some advice, not sure I'd recommend making a 20 inch x 20 inch figure. If you're having issues with your figure showing up too small in the display window, it might be better to increase the dpi with something like `pplt.rc['figure.dpi'] = 300` or passing e.g. `pplt.figure(..., dpi=300)` . That way the labels are still readable.

**Comment by Marston** [comment]
OK. I have a conda env and doing pip install always corrupts my env. I'll wait for the release. 
I understood that the warning was not critical. I just wanted to understand what it was saying. What was the cause?
As for the third issue, I tested a cartopy= matplotlib and didn't get the problem with proplot. 

`
titles = ['Eval 210930 Proxy','Baseline 210930 Proxy','Eval - Baseline 210930 Proxy']
fig, (ex, ix, dx) = plt.subplots(nrows=3, figsize=(12,12), subplot_kw={'projection':ccrs.PlateCarree()})

for ax in [ex, ix, dx]:
    ax.coastlines()
    ax.gridlines()
    ax.label_outer()

# Plot the data    
shrink = 1.0
cmap = 'nipy_spectral_r'#'gist_ncar_r' #'gist_stern_r'
ev.plot(ax=ex, transform=ccrs.PlateCarree(),cmap=cmap, vmin=0.1, vmax=20, cbar_kwargs={'shrink': shrink})
ib.plot(ax=ix, transform=ccrs.PlateCarree(),cmap=cmap, vmin=0.1, vmax=20, cbar_kwargs={'shrink': shrink})
dif.plot(ax=dx, transform=ccrs.PlateCarree(),cmap='seismic', vmin=-15, vmax=15, cbar_kwargs={'shrink': shrink})

for ax, title in zip([ex, ix, dx],titles):
    ax.set_title(title)
    
ev.close()
ib.close()
dif.close()
`

The above code produces the proper treatment of the dataset in this fig:

![baseline-eval-m-c](https://user-images.githubusercontent.com/1478822/136477753-4fa3f390-a10f-4901-b47a-c4bbe96eb436.png)


Thanks for the figsize tip.

**Comment by lukelbd** [comment]
Yeah `pip install` sometimes messes up the conda environment, but it's pretty safe with proplot since the dependencies are unusually simple (literally just `matplotlib`). The warning was caused by proplot trying to apply the "edgefix" feature to `AxesImage` objects returned by `pcolorfast`, despite the fact that `AxesImage` objects don't trigger the issue where white lines appear between vector graphic shading regions (see the [pcolor docs](https://proplot.readthedocs.io/en/latest/api/proplot.axes.PlotAxes.pcolor.html#proplot.axes.PlotAxes.pcolor) for details -- keyword is `edgefix`).

As for the contour issue, it looks like your example with just matplotlib + cartopy might still be presenting the problem -- except your colormap is different, so the whole background is getting colored by white instead of light blue. I've been able to reproduce this problem in just cartopy + matplotlib (no proplot), so it is indeed a cartopy problem:

```python
import numpy as np
import matplotlib.pyplot as plt
import cartopy.crs as cars
N = 10
state = np.random.RandomState(23)
fig = plt.figure(figsize=(5, 2.5))
ax = fig.add_subplot(projection=ccrs.Mollweide())
ax.coastlines()
x = np.linspace(-180, 180, N)
y = np.linspace(-90, 90, N)
z = state.rand(N, N) * 10 - 5
m = ax.contourf(
    x, y, z,
    transform=pplt.PlateCarree(),
    edgefix=False,
    cmap='NegPos',
    vmin=-5, vmax=5,
)
fig.colorbar(m, ax=ax)
```

![bug](https://user-images.githubusercontent.com/19657652/136681791-b736bad4-e51e-43b9-9b3b-bdaf4a46470b.png)

I think I'll submit a bug report to cartopy with this example.
